compose/service.py

```diff
@@ -605,7 +605,9 @@ def _get_container_host_config(self, override_options, one_off=False):
     def build(self, no_cache=False):
         log.info('Building %s...' % self.name)
 
-        path = six.binary_type(self.options['build'])
+        path = self.options['build']
+        if not six.PY3:
+            path = path.encode('utf8')
```

Comment:
Note to self: add a comment about why this branch  exists (working around python2 os.path not supporting unicode)